internal/command/updateimagetag.go

```diff
@@ -0,0 +1,190 @@
+			}
+			clean, err := gitrepo.IsClean(ctx, apiRepo)
+			if err != nil {
+				return err
+			}
+			if !clean {
+				return errors.New("api repo must be clean before updating the language image tag")
+			}
+		}
+
+		var outputDir string
+		if flagOutput == "" {
+			outputDir = filepath.Join(tmpRoot, "output")
+			if err := os.Mkdir(outputDir, 0755); err != nil {
+				return err
+			}
+			slog.Info(fmt.Sprintf("No output directory specified. Defaulting to %s", outputDir))
+		} else {
+			outputDir, err = filepath.Abs(flagOutput)
+			if err != nil {
+				return err
+			}
+		}
+
+		var languageRepo *gitrepo.Repo
+		if flagRepoRoot == "" {
+			languageRepo, err = cloneLanguageRepo(ctx, flagLanguage, tmpRoot)
+			if err != nil {
+				return err
+			}
+		} else {
+			repoRoot, err := filepath.Abs(flagRepoRoot)
+			if err != nil {
+				return err
+			}
+			languageRepo, err = gitrepo.Open(ctx, repoRoot)
+			if err != nil {
+				return err
+			}
+			clean, err := gitrepo.IsClean(ctx, apiRepo)
+			if err != nil {
+				return err
+			}
+			if !clean {
+				return errors.New("language repo must be clean before update")
+			}
+		}
+
+		state, err := loadState(languageRepo)
+		if err != nil {
+			return err
+		}
+
+		if state.ImageTag == flagTag {
+			return errors.New("specified tag is already in language repo state")
+		}
+		state.ImageTag = flagTag
+		image := deriveImage(state)
+		saveState(languageRepo, state)
+
+		// Take a defensive copy of the generator input directory from the language repo.
+		generatorInput := filepath.Join(tmpRoot, "generator-input")
+		if err := os.CopyFS(generatorInput, os.DirFS(filepath.Join(languageRepo.Dir, "generator-input"))); err != nil {
+			return err
+		}
+
+		// Perform "generate, clean" on each element in ApiGenerationStates.
+		for _, apiState := range state.ApiGenerationStates {
+			err = regenerateApi(ctx, apiRepo, languageRepo, generatorInput, image, outputDir, apiState)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Commit any changes
+		commitMsg := fmt.Sprintf("chore: update generation image tag to %s", flagTag)
+		if err := commitAll(ctx, languageRepo, commitMsg); err != nil {
+			return err
+		}
+
+		// Build everything at the end. (This is more efficient than building each library with a separate container invocation.)
+		slog.Info("Building all libraries.")
+		if err := container.BuildLibrary(image, languageRepo.Dir, ""); err != nil {
+			return err
+		}
+
+		if !flagPush {
+			slog.Info("Pushing not specified; update complete.")
+			return nil
+		}
+
+		return push(ctx, languageRepo, startOfRun, "chore: update generation image tag", "")
+	},
+}
+
+func regenerateApi(ctx context.Context, apiRepo *gitrepo.Repo, languageRepo *gitrepo.Repo, generatorInput string, image string, outputRoot string, apiState *statepb.ApiGenerationState) error {
+	if err := gitrepo.Checkout(ctx, apiRepo, apiState.LastGeneratedCommit); err != nil {
+		return err
+	}
+	slog.Info(fmt.Sprintf("Generating '%s'", apiState.Id))
```

Comment:
Just want to confirm my understanding: I might missed this in the docs, is "AutomationLevel" only affect [updateapi](https://github.com/googleapis/librarian/blob/aa7b45bc089c835ca90ac8125e28aa19e31fe9b7/internal/command/updateapis.go#L175-L178), but not updateimagetag? iiuc, this means even if automation level is blocked, we still want to update the generated code for generator change, just not api proto changes?

Reply:
Yes, correct. Even if an API is blocked, the commit should refer to the last *successful* generation, so we'd expect it to be okay with the new image.